2018 Robot/src/org/usfirst/frc/team4999/robot/controllers/WheelWrapper.java

```diff
@@ -0,0 +1,74 @@
+package org.usfirst.frc.team4999.robot.controllers;
+
+import org.usfirst.frc.team4999.robot.RobotMap;
+
+import edu.wpi.first.wpilibj.Joystick;
+
+public class WheelWrapper extends DriveController {
+	
+	private Joystick wheel = RobotMap.wheel;
+	
+	private static final double[] Gears = {0.2, 0.4, 0.6, 0.8, 1};
+	private int currentGear = 2;
+	
+	private int liftSpeed = 1;
+
+	@Override
+	public double getMoveRequest() {
+		double mr = -wheel.getRawAxis(2);
+		return map(mr, -1, 1, 0, 1);
+	}
+
+	@Override
+	public double getTurnRequest() {
+		return wheel.getRawAxis(0);
+	}
+
+	@Override
+	public double getSpeedLimiter() {
+		if(wheel.getRawButtonPressed(6) && currentGear < Gears.length-1)
+			currentGear++;
```

Comment:
`getButtonPressed()` only returns true if [the button was pressed since the last check](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/GenericHID.html). I've tested this to work as desired